GitCommands/Git/GitGpgController.cs

```diff
@@ -0,0 +1,297 @@
+
+                _revision = value;
+
+                NumberOfTag = _revision.Refs.Count(x => x.IsTag && x.IsDereference);
+
+                TagVerifyMessage = "";
+            }
+        }
+
+        /// <summary>
+        /// Accessor for Module property.
+        /// </summary>
+        public IGitModule Module
+        {
+            get
+            {
+                return _module;
+            }
+
+            set
+            {
+                if (_module == null)
+                {
+                    throw new ArgumentNullException("module");
+                }
+
+                _module = value;
+            }
+        }
+
+
+        /// <summary>
+        /// Obtain the commit signature status on current revision.
+        /// </summary>
+        /// <returns>Enum value that indicate the gpg status for current git revision.</returns>
+        public CommitStatus CheckCommitSign()
+        {
+            CheckVariable();
+
+            CommitStatus _cmtStatus = CommitStatus.NoSignature;
+            string _gpg = _module.RunGitCmd($"log --pretty=\"format:%G?\" -1 {_revision.Guid}");
+
+            switch (_gpg)
+            {
+                case GOOD_SIGN:         // "G" for a good (valid) signature
+                    _cmtStatus = CommitStatus.GoodSignature;
+                    break;
+                case BAD_SIGN:          // "B" for a bad signature
+                case UNK_SIGN_VALIDITY: // "U" for a good signature with unknown validity 
+                case EXPIRED_SIGN:      // "X" for a good signature that has expired
+                case EXPIRED_SIGN_KEY:  // "Y" for a good signature made by an expired key
+                case REVOKED_KEY:       // "R" for a good signature made by a revoked key
+                    _cmtStatus = CommitStatus.SignatureError;
+                    break;
+                case MISSING_PUB_KEY:   // "E" if the signature cannot be checked (e.g.missing key)
+                    _cmtStatus = CommitStatus.MissingPublicKey;
+                    break;
+                case NO_SIGN:           // "N" for no signature
+                default:
+                    _cmtStatus = CommitStatus.NoSignature;
+                    break;
+            }
+
+            return _cmtStatus;
+        }
+
+
+        /// <summary>
+        /// Obtain the tag status on current revision.
+        /// </summary>
+        /// <returns>Enum value that indicate if current git revision has one tag with good signature, one tag with bad signature or more than one tag.</returns>
+        public TagStatus CheckTagSign()
+        {
+            CheckVariable();
+
+            TagStatus _tagStatus = TagStatus.OneBad;
+
+            IEnumerable<IGitRef> _usefulRef = _revision.Refs.Where(x => x.IsTag && x.IsDereference);
+
+            Regex goodSignatureTagRegex = new Regex(_goodSignature);
+            Match goodSignatureMatch = null;
+
+            Regex validSignatureTagRegex = new Regex(_validTagSign);
```

Comment:
Aren't these regexes static? 
Worth changing to `new Regex(_validTagSign, RegexOptions.Compiled);` too

Reply:
Yes they are fixed, so they can be created as you wrote. 
I did not know about `RegexOptions.Compiled` parameter.

Reply:
In this case it is better to move them out of the method and make static variables.
This should improve performance

Reply:
Done.